src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.PackTool.targets

```diff
@@ -0,0 +1,49 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.PackTool.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="Microsoft.NET.Build.Tasks.GenerateToolsSettingsFile"
+              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <Target Name="PackTool" DependsOnTargets="GenerateToolsSettingsFileFromBuildProperty;Publish" Condition=" '$(PackAsTool)' == 'true' ">
+    <!--All published file in different tools/tfm-->
+
+    <ItemGroup>
+        <_PublishedTfmSpecificPackageFiles Include="@(ResolvedFileToPublish->'$(PublishDir)%(RelativePath)')"/>
+        <_PublishedTfmSpecificPackageFiles Include="$(PublishDepsFilePath)"/>
+        <_PublishedTfmSpecificPackageFiles Include="$(PublishRuntimeConfigFilePath)"/>
+    </ItemGroup>
+
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(_PublishedTfmSpecificPackageFiles)">
+        <PackagePath>tools/$(targetframework)/any/%(_PublishedTfmSpecificPackageFiles.RecursiveDir)%(_PublishedTfmSpecificPackageFiles.Filename)%(_PublishedTfmSpecificPackageFiles.Extension)</PackagePath>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+
+    <ItemGroup>
+       <PackageReference Remove="@(PackageReference)"/>
+    </ItemGroup>
+  </Target>
+
+  <PropertyGroup>
+    <ToolCommandName Condition=" '$(ToolCommandName)' == '' ">$(TargetName)</ToolCommandName>
+    <ToolEntryPoint Condition=" '$(ToolEntryPoint)' == '' ">$(TargetFileName)</ToolEntryPoint>
+  </PropertyGroup>
+
+  <Target Name="GenerateToolsSettingsFileFromBuildProperty" >
+    <GenerateToolsSettingsFile
+       EntryPointRelativePath="$(ToolEntryPoint)"
+       CommandName="$(ToolCommandName)"
```

Comment:
Is there only ever going to be one entry point? If this task took multiple inputs via items with metadata instead of the properties, one could extend the pack by smuggling in additional runnable assemblies and adding items..

Reply:
It is still in design. As of today in master, the setting format allow multiple entry point. However, CLI is blocking more than one entry point, due to time, no clear use case and no design on conflict (Package 1 has tool A,B,**C** and Package 2 has tool **C**, D, E).

I am still focus on the basic scenario. And for pack, because I am defaulting to `TargetFileName` which is only one, it is easier for the producer to understand "override the default" than "you can have multiple, and also you can keep the default and adding more or remove the default and adding more". The interface will also need to be more complex.